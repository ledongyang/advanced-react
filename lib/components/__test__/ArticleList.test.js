import React from 'react';
import ArticelList from '../ArticleList';

import renderer from 'react-test-renderer';

describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: {
        id: 'a',
        title: 'a',
        body: 'a',
        date: 'a'
      },
      b: {
        id: 'b',
        title: 'b',
        body: 'b',
        date: 'b'
      },
      c: {
        id: 'c',
        title: 'c',
        body: 'c',
        date: 'c'
      }
    },
    store: {
      lookUpAuthor: jest.fn(() => ({}))
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <ArticelList
        {...testProps}
      />
    ).toJSON();

    expect(tree.children.length).toBe(3);

    expect(tree).toMatchSnapshot();
  });
});
