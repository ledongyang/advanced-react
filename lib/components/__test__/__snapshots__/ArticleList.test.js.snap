// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      a
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      a
    </div>
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      b
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      b
    </div>
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      c
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      c
    </div>
  </div>
</div>
`;
